### DISCUSSION TITLE
Instance constraints of a state variable depending on the values of other state variables.

### DISCUSSION THREAD

**Discussion by Peter230655** [csu-hmc/opty#264] [Ideas]
I am presently playing around with opty trying to solve test problems in John T. Betts' book _Practical Methods for Optimal Control Using Nonlinear Programming, 3rd edition.
In example 10.73 the instance constraint for one state variable depends on the values on other state valiables at the same time. In example 10.45 the instance constraint of one state variable depends on the value of another state variable at a different time.
As I understand opty's documentation this is presently not possible. 

If my understanding is correct, maybe a nice feature to have - if my understanding is incorrect, I'd like to know how to do it.
**Thanks!**

**Comment by moorepants** [comment]
You should be able to make instance constraints like `x(1.0) - y(1.0)`. I think opty should handle that correctly. But maybe it doesn't. In #29 I need such a constraint for periodicity but I'm not sure it was working there.

**Reply by Peter230655** [reply]
I will try and report. Would then something like `x(0) - y(1)` also work?

**Reply by moorepants** [reply]
I think it should, but I'm not sure.

**Reply by Peter230655** [reply]
YES, it works.
I tried with betts_10_50 example, for which I made a PR some time ago.
I will change the example to reflect this and push it.
